internal/poadraftapplication/db.go

```diff
@@ -0,0 +1,89 @@
+package poadraftapplication
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/ministryofjustice/opg-search-service/internal/index"
+)
+
+type DB struct {
+	conn *pgx.Conn
+}
+
+type rowResult struct {
+	ID int
+	UID string
+	donorname string
+	donorpostcode string
+	donordob string
+}
+
+func NewDB(conn *pgx.Conn) *DB {
+	return &DB{conn: conn}
+}
+
+func (db *DB) QueryIDRange(ctx context.Context) (min int, max int, err error) {
+	return min, max, db.conn.QueryRow(ctx, "SELECT COALESCE(MIN(id), 0), COALESCE(MAX(id), 0) FROM poa.draft_applications").Scan(&min, &max)
+}
+
+func (db *DB) QueryByID(ctx context.Context, results chan<- index.Indexable, from, to int) error {
+	fields := []string{"uid", "donorname", "donordob", "donorpostcode"}
+
+	query := `SELECT id, COALESCE(` + strings.Join(fields, ", ''), COALESCE(") + `, '')
```

Comment:
Could we type this out in full rather than use the join? It took me a while to understand it and I'm always wary of having any string manipulation in a SQL statement